### ISSUE TITLE
Deterministic AI ?

### ISSUE THREAD

**Issue by vincent-grosbois** [ihhub/fheroes2#4268]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

It seems that currently the AI is fully random
Shouldn't it be the case that it's ""deterministic"" like for the battle ?
At least as an option, to make debugging easier
Otherwise it seems it would get difficult to reproduce previous bugs

### Describe the possible solution

deterministic AI

**Comment by oleg-derevenetz** [comment]
Hi @vincent-grosbois according to #4265 there is a desire to make the game fully deterministic. I'm personally against of this (I still think that random events should be really random), but in general why not, if there will be the switch to turn on the "fully random" behavior.

**Comment by vincent-grosbois** [comment]
> Hi @vincent-grosbois according to #4265 there is a desire to make the game fully deterministic. I'm personally against this (I still think that random events should be really random, and poorly designed random generator in OG is not an excuse), but in general why not, if there will be the switch to turn on the "fully random" behavior.

I agree for the AI (even though that is debatable), but to the upside of this is really to simplify debugging of AI behavior

**Comment by oleg-derevenetz** [comment]
@zenseii

> Is this issue still relevant or did we close it by making deterministic AI the default in https://github.com/ihhub/fheroes2/pull/6172 ?

These are different things. Deterministic battles means (among other things, like the deterministic random events) deterministic AI only for battles, but not in general.

**Comment by monuszko** [comment]
There's still one place where AI appears random on adventure map: when using teleporters.

To my surprise, I discovered that pressing "space" while on a teleporter doesn't consume any movement points! So you can keep pressing Space (or clicking the "rearing horse" icon) to try all the unblocked destination points. But computer opponent doesn't act that way. He seems to exit at a random destination and do its actions there.

I think AI should try all the available teleport exits first, and then choose the most desirable exit.
